app/models/support/requests/time_constraint.rb

```diff
@@ -4,11 +4,27 @@ module Support
   module Requests
     class TimeConstraint
       include ActiveModel::Model
-      attr_accessor :not_before_date, :not_before_time, :needed_by_date, :needed_by_time, :time_constraint_reason
+      attr_accessor :not_before_date, :not_before_time, :needed_by_date, :needed_by_time, :time_constraint_reason, :is_legal_deadline
+
+      validates :is_legal_deadline, inclusion: %w[yes no], allow_blank: true
 
       validates_date :needed_by_date, allow_nil: true, allow_blank: true, on_or_after: :today
       validates_date :not_before_date, allow_nil: true, allow_blank: true, on_or_after: :today
       validates_date :not_before_date, on_or_before: :needed_by_date, unless: proc { |c| c.needed_by_date.blank? || c.not_before_date.blank? }
+
+      validates_time :not_before_time, allow_nil: true, allow_blank: true
+      validates_time :needed_by_time, allow_nil: true, allow_blank: true
```

Comment:
What happens if the user doesn't fill in the field? I imagine we only need to `allow_nil` or `allow_blank`, not both (probably the latter?).